core/src/edu/rice/rbox/Replication/ReplicaManagerGrpc.java

```diff
@@ -158,13 +158,18 @@ public void handleUnsubscribe(Rbox.UnsubscribeRequest request, StreamObserver<Em
         public void handleUpdate(Rbox.UpdateMessage request, StreamObserver<Empty> responseObserver) {
             logger.log(Level.INFO, "Handling update...");
 
+            RemoteChange change = getRemoteChangeFromUpdateMessage(request);
+
             // Call ChangeReceiver
-            // TODO: Cannot reference receiveChange from static context?
-            ChangeReceiver.receiveChange(request.getRemoteChange());
+            changeReceiver.receiveChange(change);
 
-            sendToReplicaHolders(request.getTargetObject(), request.getRemoteChange());
+            // Check if secondary, and if so notify registrar
+            if (secondaries.contains(change.getTarget())) {
+                // TODO: Implement once registrar interface exists
+            }
 
-            // TODO: Do we need to send a response?
+            // Check if deleting replica and if so update publishers
+            // TODO: How do we check if change is DeleteReplicaChange?
```

Comment:
Maybe research on whether serializing and deserializing would retain the subclass, would `if (change instanceof DeleteReplicaChange)` work?